tests/api-tests/queries/cursor-pagination.test.ts

```diff
@@ -0,0 +1,243 @@
+        author: relationship({ ref: 'User.posts', many: true }),
+      },
+    }),
+    User: list({
+      access: allowAll,
+      fields: {
+        name: text(),
+        posts: relationship({ ref: 'Post.author', many: true }),
+      },
+    }),
+  },
+});
+
+describe('cursor pagination basic tests', () => {
+  let testEnv: TestEnv<TypeInfoFromConfig<typeof config>>;
+  let context: KeystoneContext;
+  let posts: { id: string }[];
+  let userId: string;
+
+  beforeAll(async () => {
+    testEnv = await setupTestEnv({ config });
+    context = testEnv.testArgs.context;
+
+    await testEnv.connect();
+
+    const result = await context.query.User.createOne({
+      data: {
+        name: 'Test',
+        posts: {
+          create: Array.from(Array(15).keys()).map(num => ({ order: num })),
+        },
+      },
+      query: 'id posts { id order }',
+    });
+    userId = result.id;
+    // posts will be added in random sequence, so need to sort by order
+    posts = result.posts.sort((a: { order: number }, b: { order: number }) => a.order - b.order);
+  });
+  afterAll(async () => {
+    await testEnv.disconnect();
+  });
+
+  test('cursor pagination test (graphql api)', async () => {
+    const { errors, data } = await context.graphql.raw({
+      query: `query { posts(
+          take: 6,\
+          skip: 1,\
+          cursor: { order: 5 }\
+          orderBy: { order: asc }\
+        ) { id order }\
+      }`,
+    });
+    expect(errors).toEqual(undefined);
+    let currentOrder = 6;
+    expect(data).toEqual({
+      posts: Array.from(Array(6).keys()).map(_ => posts[currentOrder++]),
+    });
+  });
+
+  test('cursor pagination test (query api)', async () => {
+    const result1 = await context.query.Post.findMany({
+      take: 6,
+      skip: 1,
+      cursor: { order: 5 },
+      orderBy: { order: 'asc' },
+      query: 'id order',
+    });
+    expect(result1).toBeDefined();
+    expect(result1.length).toBe(6);
+    let currentOrder = 6;
+    expect(result1).toEqual(Array.from(Array(6).keys()).map(_ => posts[currentOrder++]));
+  });
+
+  test('cursor pagination test (db api)', async () => {
+    const result1 = await context.db.Post.findMany({
+      take: 6,
+      skip: 1,
+      cursor: { order: 5 },
+      orderBy: { order: 'asc' },
+    });
+    expect(result1).toBeDefined();
+    expect(result1.length).toBe(6);
+    let currentOrder = 6;
+    expect(result1).toEqual(Array.from(Array(6).keys()).map(_ => posts[currentOrder++]));
+  });
+
+  test('cursor pagination through relation', async () => {
+    const { errors, data } = await context.graphql.raw({
+      query: `query {\
+        user(where: { id: "${userId}"}) {\
+          posts(\
+            take: 6,\
+            skip: 1,\
+            cursor: { order: 5 }\
+            orderBy: { order: asc }\
+          ) { id order }\
+        }\
+      }`,
+    });
+    expect(errors).toEqual(undefined);
+    let currentOrder = 6;
```

Comment:
We should alter this to use an `autoincrement` identifier rather than something named `order`, but we won't block merging on that